src/scripts/main.js

```diff
@@ -1,3 +1,65 @@
 'use strict';
 
 // write code here
+const table = document.querySelector('table');
+const headerOfTable = table.querySelector('thead');
+
+if (headerOfTable) {
+  headerOfTable.addEventListener('click', (e) => {
+    const titles = [...headerOfTable.querySelectorAll('th')];
+    const clickedHeader = e.target.closest('th');
+
+    if (!clickedHeader) {
+      return;
+    }
+
+    const tableData = [...table.querySelectorAll('tbody tr')].map((row) => {
+      return [...row.querySelectorAll('td')].map((cell) => {
+        return cell.textContent.trim();
+      });
+    });
+
+    const index = titles.indexOf(e.target);
```

Comment:
The method `indexOf` is used incorrectly here. `e.target` is the clicked element, but `titles` is an array of `th` elements. You should use `titles.indexOf(clickedHeader)` instead to get the correct index of the clicked header.